apps/visualization/schema.py

```diff
@@ -81,6 +84,43 @@ async def get_country_profile_object(iso3):
         return None
 
 
+@sync_to_async
+def get_sources(iso3, emergency, indicator_id, indicator_name, indicator_description, subvariable, thematic, topic, type):
+    filters = clean_filters({
+        'iso3': iso3,
+        'emergency': emergency,
+        'indicator_id': indicator_id,
+        'indicator_name': indicator_name,
+        'indicator_description': indicator_description,
+        'subvariable': subvariable,
+        'thematic': thematic,
+        'topic': topic,
+        'type': type,
+    })
+    qs = DataGranular.objects.filter(**filters)
+    if qs:
```

Comment:
This is always true.

Reply:
@thenav56 , May be it would be fine if we use qs.exists()

Reply:
Maybe we can just use the latest_date.
```python3
....
latest_data_granular = qs.latest('indicator_month')
if not latest_data_granular:
    return []
data = qs.filter(indicator_month=indicator_month.latest_date).values(
.....
```